packages/js/src/core/fetch-token.ts

```diff
@@ -0,0 +1,75 @@
+import { KeysToCamelCase } from '@silverhand/essentials';
+import { string, number, assert, type, StructError, Infer } from 'superstruct';
+import { Except } from 'type-fest';
+
+import { LogtoError } from '../utils/errors';
+import { createRequester } from '../utils/requester';
+
+export interface FetchTokenByRefreshTokenParameters {
+  clientId: string;
+  tokenEndPoint: string;
+  refreshToken: string;
+  resource?: string;
+  scope?: string[];
+}
+
+const TokenResponseSchema = type({
+  access_token: string(),
+  refresh_token: string(),
+  id_token: string(),
+  scope: string(),
```

Comment:
I am not quite sure whether the scope return value is required or not. If no scope if provided, I believe it could be undefined. Let's go with the optional for now. I'll update the doc.

Reply:
ok, makes sense to me